### ISSUE TITLE
[PPML] Cluster Serving job depends on scala

### ISSUE THREAD

**Issue by qiyuangong** [intel-analytics/BigDL#3352]
```bash
java.lang.NoSuchMethodError: 'void scala.Product.$init$(scala.Product)'
        at scopt.OptionParser.<init>(options.scala:226)
        at com.intel.analytics.bigdl.serving.ClusterServing$$anon$1.<init>(ClusterServing.scala:40)
        at com.intel.analytics.bigdl.serving.ClusterServing$.<init>(ClusterServing.scala:40)
        at com.intel.analytics.bigdl.serving.ClusterServing$.<clinit>(ClusterServing.scala)
        at com.intel.analytics.bigdl.serving.ClusterServing.main(ClusterServing.scala)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:288)
        at org.apache.flink.client.program.PackagedProgram.invokeInteractiveModeForExecution(PackagedProgram.java:198)
        at org.apache.flink.client.ClientUtils.executeProgram(ClientUtils.java:149)
        at org.apache.flink.client.cli.CliFrontend.executeProgram(CliFrontend.java:699)
        at org.apache.flink.client.cli.CliFrontend.run(CliFrontend.java:232)
        at org.apache.flink.client.cli.CliFrontend.parseParameters(CliFrontend.java:916)
        at org.apache.flink.client.cli.CliFrontend.lambda$main$10(CliFrontend.java:992)
        at org.apache.flink.runtime.security.contexts.NoOpSecurityContext.runSecured(NoOpSecurityContext.java:30)
        at org.apache.flink.client.cli.CliFrontend.main(CliFrontend.java:992)
```

**Comment by glorysdj** [comment]
need to package scala dependencies in the with-dependencies jar
or include scala jars in the classpath of the cluster serving job

**Comment by glorysdj** [comment]
BTW, do we have a cluster serving test job on Jenkins?

**Comment by Litchilitchy** [comment]
could not reproduce with `bigdl-serving-jar-with-dependencies`

also, with `jar tvf` I could see `scala/Product` in jar, could you see what's in your jar, and which jar are you using?

**Comment by glorysdj** [comment]
will update ppml docker to bigdl-serving-spark_2.4.6_0.14.0-SNAPSHOT-jar-with-dependencies

**Comment by Litchilitchy** [comment]
If use spark 3.x, scala version would be set to 2.12, thus flink client should be flink_1.11.3_scala_2.11 as well

**Comment by Litchilitchy** [comment]
> If use spark 3.x, scala version would be set to 2.12, thus flink client should be flink_1.11.3_scala_2.12 as well

**Comment by qiyuangong** [comment]
> If use spark 3.x, scala version would be set to 2.12, thus flink client should be flink_1.11.3_scala_2.12 as well

Do it means we need to add scala version for flink download?
Such as flink-${FLINK_VERSION}-bin-${SCALA_VERSION}.tgz

**Comment by Litchilitchy** [comment]
> > If use spark 3.x, scala version would be set to 2.12, thus flink client should be flink_1.11.3_scala_2.12 as well
> 
> Do it means we need to add scala version for flink download? Such as flink-${FLINK_VERSION}-bin-${SCALA_VERSION}.tgz

I think yes

**Comment by qiyuangong** [comment]
Issue resolved by #3363
